src/cim/components/InfraEnv/InfraEnvAgentTable.tsx

```diff
@@ -1,22 +1,61 @@
 import * as React from 'react';
+import { DropdownItem } from '@patternfly/react-core';
+import { noop } from 'lodash';
+
 import { Host } from '../../../common/api/types';
 import {
   discoveryTypeColumn,
   infraEnvStatusColumn,
   clusterColumn,
   useAgentsTable,
 } from '../Agent/tableUtils';
-import HostsTable, { HostsTableEmptyState } from '../../../common/components/hosts/HostsTable';
+import HostsTable, {
+  DefaultExpandComponent,
+  HostsTableEmptyState,
+} from '../../../common/components/hosts/HostsTable';
 import {
   cpuCoresColumn,
   discoveredAtColumn,
   disksColumn,
   hostnameColumn,
   memoryColumn,
 } from '../../../common/components/hosts/tableUtils';
-import { DiscoveryTroubleshootingModal } from '../../../common';
+import {
+  DiscoveryTroubleshootingModal,
+  HostToolbar,
+  ChangeHostnameAction,
+  MassChangeHostnameModal,
+} from '../../../common';
 import { TableRow } from '../../../common/components/hosts/AITable';
 import { InfraEnvAgentTableProps } from '../ClusterDeployment/types';
+import MassApproveAgentModal from '../modals/MassApproveAgentModal';
+import { ActionItemsContext } from '../../../common/components/hosts/HostToolbar';
+import { AgentK8sResource } from '../../types';
+
+type MassApproveActionProps = {
+  onApprove: VoidFunction;
+  selectedAgents: AgentK8sResource[];
+};
+
+const MassApproveAction: React.FC<MassApproveActionProps> = ({ onApprove, selectedAgents }) => {
+  const isDisabled = React.useContext(ActionItemsContext);
+  const approveAvailable = selectedAgents.some((a) => !a.spec.approved);
+  return (
+    <DropdownItem
+      onClick={onApprove}
+      isDisabled={isDisabled || !approveAvailable}
+      description={
+        isDisabled
+          ? 'Select one or more hosts to approve'
```

Comment:
This is hard to read. Can we avoid chaining of the ternary operator? And Move the logic upwards.

Reply:
fixed